Cannot use `match` for byte arrays (even in variables)

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 6b683d3.465860e
**OS:** linux, "Manjaro Linux"

**What did you do?**
```
fn check_signature(signature []byte) byte {
	gim_little_endian := [byte(0x4D), 0x49, 0x47, 0x2E, 0x30, 0x30, 0x2E, 0x31]
	gim_big_endian := [byte(0x2E), 0x47, 0x49, 0x4D, 0x31, 0x2E, 0x30, 0x30]
	return match signature {
		gim_little_endian { 1 }
		gim_big_endian { 2 }
		else { 0 }
	}
}

fn main() {
	arr := [byte(0x4D), 0x49, 0x47, 0x2E, 0x30, 0x30, 0x2E, 0x31]
	println(check_signature(arr))
}
```

**What did you expect to see?**
`1`
 
**What did you see instead?**
```
==================
/tmp/v/check_signature.11746897608754086962.tmp.c:10082: error: invalid operand types for binary operation
/tmp/v/check_signature.11746897608754086962.tmp.c:10082: error: invalid aggregate type for register load
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4231,6 +4231,27 @@ fn (mut g Gen) match_expr_classic(node ast.MatchExpr, is_expr bool, cond_var str
 						g.expr(expr)
 						g.write(')')
 					}
+				} else if type_sym.kind == .array {
+					ptr_typ := g.gen_array_equality_fn(node.cond_type)
+					g.write('${ptr_typ}_arr_eq(')
+					g.write(cond_var)
+					g.write(', ')
+					g.expr(expr)
+					g.write(')')
+				} else if type_sym.kind == .array_fixed {
+					ptr_typ := g.gen_fixed_array_equality_fn(node.cond_type)
+					g.write('${ptr_typ}_arr_eq(')
+					g.write(cond_var)
+					g.write(', ')
+					g.expr(expr)
+					g.write(')')
+				} else if type_sym.kind == .map {
+					ptr_typ := g.gen_map_equality_fn(node.cond_type)
+					g.write('${ptr_typ}_map_eq(')
+					g.write(cond_var)
+					g.write(', ')
+					g.expr(expr)
+					g.write(')')
 				} else if expr is ast.RangeExpr {
 					// if type is unsigned and low is 0, check is unneeded
 					mut skip_low := false

--- a/vlib/v/tests/match_array_or_map_cond_test.v
+++ b/vlib/v/tests/match_array_or_map_cond_test.v
@@ -1,0 +1,37 @@
+fn test_match_array_or_map_cond() {
+	// array
+	y1 := []byte{}
+	x1 := []byte{}
+	ret1 := match x1 {
+		y1 { true }
+		else { false }
+	}
+	println('ret1 is $ret1')
+	assert ret1
+
+	// fixed array
+	y2 := [1, 2, 3]!
+	x2 := [1, 2, 3]!
+	ret2 := match x2 {
+		y2 { true }
+		else { false }
+	}
+	println('ret2 is $ret2')
+	assert ret2
+
+	// map
+	y3 := map{
+		1: 11
+		2: 22
+	}
+	x3 := map{
+		1: 11
+		2: 22
+	}
+	ret3 := match x3 {
+		y3 { true }
+		else { false }
+	}
+	println('ret3 is $ret3')
+	assert ret3
+}

